example/googleMapsExample.js

```diff
@@ -0,0 +1,604 @@
+
+	fetch( url, { mode: 'cors' } )
+		.then( ( res ) => {
+
+			if ( res.ok ) {
+
+				return res.json();
+
+			} else {
+
+				return Promise.reject( `${res.status} : ${res.statusText}` );
+
+			}
+
+		} )
+		.then( ( json ) => {
+
+			if ( !json.root ) {
+				throw new Error( `malformed response: ${ json }` );
+			}
+
+			// TODO: check this is correct
+			// find first node with uri and treat that as root
+			let uri = undefined;
+			const toVisit = [];
+			for ( let curr = json.root; curr !== undefined; curr = toVisit.pop() ) {
+				if ( curr.content?.uri ) {
+					uri = new URL( `${ apiOrigin }${ curr.content.uri }` );
+					uri.searchParams.append( 'key', params.apiKey );
+					break;
+				}
+
+				toVisit.push( ...curr.children );
+			}
+
+			if ( !uri ) {
+				throw new Error( `can't find root uri in response: ${ json }` );
+			}
+
+			const session = uri.searchParams.get( 'session' );
+
+			console.log('uri', uri);
+			console.log('session', session);
+
+			tiles = new TilesRenderer( uri.toString() );
+
+			tiles.preprocessURL = uri => {
+
+				uri = new URL( uri );
+				if ( /^http/.test( uri.protocol ) ) {
+
+					uri.searchParams.append( 'session', session );
+					uri.searchParams.append( 'key', params.apiKey );
+
+				}
+				return uri.toString();
+
+			};
+
+			tiles.onLoadTileSet = () => {
+
+				const box = new Box3();
+				const sphere = new Sphere();
+				const matrix = new Matrix4();
+
+				let position;
+				let distanceToEllipsoidCenter;
+
+				if ( tiles.getOrientedBounds( box, matrix ) ) {
+
+					position = new Vector3().setFromMatrixPosition( matrix );
+					distanceToEllipsoidCenter = position.length();
+
+				} else if ( tiles.getBoundingSphere( sphere ) ) {
+
+					position = sphere.center.clone();
+					distanceToEllipsoidCenter = position.length();
+
+				}
+
+				const surfaceDirection = position.normalize();
+				const up = new Vector3( 0, 1, 0 );
+				const rotationToNorthPole = rotationBetweenDirections( surfaceDirection, up );
+
+				tiles.group.quaternion.x = rotationToNorthPole.x;
+				tiles.group.quaternion.y = rotationToNorthPole.y;
+				tiles.group.quaternion.z = rotationToNorthPole.z;
+				tiles.group.quaternion.w = rotationToNorthPole.w;
+
+				tiles.group.position.y = - distanceToEllipsoidCenter;
+
+			};
+
+			setupTiles();
+
+			} )
+			.catch( err => {
+
+				console.error( 'Unable to get gmaps tileset:', err );
+
+			} );
+
+}
+
+function init() {
+	scene = new Scene();
+
+	// primary camera view
+	renderer = new WebGLRenderer( { antialias: true } );
+	renderer.setPixelRatio( window.devicePixelRatio );
+	renderer.setSize( window.innerWidth, window.innerHeight );
+	renderer.setClearColor( 0x151c1f );
+	renderer.outputEncoding = sRGBEncoding;
+
+	document.body.appendChild( renderer.domElement );
+	renderer.domElement.tabIndex = 1;
+
+	camera = new PerspectiveCamera( 60, window.innerWidth / window.innerHeight, 1, 4000 );
+	camera.position.set( 10, 10, 10 );
```

Comment:
oh of course! thanks.

Reply:
I've added some temporary positioning of the camera and it works!
![image](https://github.com/NASA-AMMOS/3DTilesRendererJS/assets/97134546/bb912b9b-b80e-4826-b2f7-80281fb3228a)
It doesn't progressively load the more detailed tiles as I zoom in. Do you have any ideas why this is, or some general pointers on how to achieve the full DIY Google Earth experience?

Reply:
Awesome!

> It doesn't progressively load the more detailed tiles as I zoom in.

First thing that comes to mind is that the LRUCache is full. I'd increase the maxSize and maybe the minSize to a significantly larger value to ensure non of the tiles are prevented from loading. The goal of the LRUCache is to prevent the page from storing too much memory without ever releasing tiles but in some cases it needs to be increased.

Something like the following:

```js
tiles.lruCache.minSize = 3000;
tiles.lruCache.maxSize = 5000;
```

Reply:
I've pushed up your suggestion - it doesn't resolve the issue. I'll dig into source code tomorrow, but maybe I'm not getting inside the bounding boxes dues to the scale of the model, and that's why the more detailed tiles aren't loading? Is that roughly how the renderer works (works out which nodes in the tileset are within some threshold of the camera position and render them)?

Reply:
I'm seeing that it is loading more tiles when zooming - but it it's trying to load tons of them because the camera frustum is encompassing the full radius of the globe. I'll look into making some adjustments that simplify and provide more of a maps like experience over the next couple days.

Reply:
I was going to make the frustum far plane progressively contract as you zoom in - do you think that'll be enough to stop the spurious loads?